### ISSUE TITLE
Printing strings and pointers fail for some cases

### ISSUE THREAD

**Issue by kkumar-fk** [iovisor/bpftrace#865]
Environment: Debian 9.8, kernel version: 4.9.144-3.1.

We @Flipkart are just starting to use EBPF to debug/trace the otherwise "un-debuggable"
issues. When using bpftrace to debug fdb_add_hw_addr(), some of the lines fail to load
as described below.

```
#include <linux/netdevice.h>
struct net_bridge
{
        spinlock_t                      lock;
        struct list_head                port_list;
        struct net_device               *dev;
}

kprobe:fdb_add_hw_addr
{
	$br = ((net_bridge *)arg0);

	// This works
	printf("FDB: br: %s addr: %p\n", $br->dev->name, arg1);

	// ERROR1: Next line does not work, gives error:
	// "Error loading program: kprobe:fdb_add_hw_addr (try -v)"
	printf("FDB: br: %s addr: %s\n", $br->dev->name, str(arg1));

	// ERROR2: Next line does not work, gives error:
	// "printf: %p specifier expects a value of type integer(cast supplied)"
	printf("Port_list.next: %p\n", $br->port_list.next);

	// ERROR3: Next line does not work, gives error:
	// 21.36: syntax error, unexpected &
	printf("Port_list Address: %p\n", &($br->port_list));
}
```

Could someone help whether this is the wrong way to do, or if it is an actual issue (printing string,
and printing pointers highlighted by ERROR2 and ERROR3 above)?

Thanks.

**Comment by kkumar-fk** [comment]
Ping.. Anyone who could help in identifying the source of the problem?

**Comment by lyveng** [comment]
For the pointer part, it looks like #252 
Looks like the fix is part of v0.9.1 as per the milestone marked in #252

**Comment by fbs** [comment]
The address of operator `&` isn't supported yet so that won't work. The rest should work in master
